### DISCUSSION TITLE
NavigateTo not working using Options API

### DISCUSSION THREAD

**Discussion by Geometrically** [nuxt/nuxt#18455] [Questions]
I am trying to use navigateTo to redirect a user on the server side if they type in a User ID vs a username. This approach worked fine in `asyncData` using Nuxt 2. However, the redirect doesn't even execute anymore (a couple days it worked, but only on the client-side, which is not ideal). I have also tried rewriting this using the `setup` function, but it still did not work. Unfortunately it would be very hard to use the composition API as this is a large project (so it would require a lot of rewriting code). 

test/[id].vue
```vue
<template>
  <div v-if="user">
    <section>User id: {{ user }}</section>
  </div>
  <div v-else></div>
</template>

<script>
export default defineNuxtComponent({
  async asyncData() {
    const route = useRoute();

    try {
      const user = await $fetch(
        `https://api.modrinth.com/v2/user/${route.params.id}`
      );
  
      if (user.username !== route.params.id) {
        await navigateTo(`/${user.username}`, { redirectCode: 301 });
        return {};
      }
    catch {
      createError({
        fatal: true,
        statusCode: 404,
        message: 'User not found',
      })
    }

    return { user: ref(user) };
  },
});
</script>
```

This code above should work, but it doesn't. If you go to `/test/modmuss50`, everything works as expected. However, when you go to `/test/TEZXhE2U`, it does not redirect, on the server or the client.

Am I doing something wrong? Is this not supported in the options API? Is this a bug?

Thanks for all your help!

**Edit**: Removing the `return` before the `navigateTo` works only on the client-side. For some reason, it still does not work on the server-side, and will return a 200.

**Comment by oritwoen** [comment]
Why this happens is described here: https://github.com/nuxt/nuxt/issues/12342#issuecomment-1400849134

**Comment by ndragun92** [comment]
Not sure if that's the same error but yea it does not work on refresh when server side was called. It works on client side when navigating
Server side redirect was working only with router.push

Example:
https://stackblitz.com/edit/github-5wgy7g?file=app.vue,pages%2Ftest%2F[id].vue,layouts%2Fdefault.vue,pages%2F[user].vue
